src/selectors/entities/schemes.js

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2018-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+// @flow
+
+import {createSelector} from 'reselect';
+import {getAllChannels} from 'selectors/entities/channels';
+import {getTeams} from 'selectors/entities/teams';
+import {ScopeTypes} from 'constants/schemes';
+import type {GlobalState} from 'types/store';
+import type {Scheme} from 'types/schemes';
+import type {Channel} from 'types/channels';
+import type {Team} from 'types/teams';
+
+export function getSchemes(state: GlobalState): { [string]: Scheme } {
+    return state.entities.schemes.schemes;
+}
+
+export function getScheme(state: GlobalState, id: string): Scheme {
+    const schemes = getSchemes(state);
+    return schemes[id];
+}
+
+export function makeGetSchemeChannels() {
+    return createSelector(
+        getAllChannels,
+        (state, props) => getScheme(state, props.schemeId),
+        (allChannels, scheme) => {
+            const {scope} = scheme;
+
+            if (scope === ScopeTypes.TEAM) {
+                const msg = `Not implemented: scheme '${scheme.id}' is team-scope but 'getSchemeChannels' only accepts channel-scoped schemes.`;
+                console.warn(msg); // eslint-disable-line no-console
+                return [];
+            }
+
+            const schemeChannels: Array<Channel> = [];
+
+            Object.entries(allChannels).forEach((item: [string, Channel]) => {
+                const [, channel: Channel] = item;
+                if (channel.scheme_id === scheme.id) {
+                    schemeChannels.push(channel);
+                }
+            });
+
+            return schemeChannels;
+        }
+    );
+}
+
+export function makeGetSchemeTeams() {
```

Comment:
I would expect to user here the getSchemeTeams action results. We can implement it this way, but then, the action is not needed. Actually, I think that is better to use the getSchemeTeams because is possible that we have a lot of teams that aren't loaded in our current context.

Reply:
The same comment can be applied to the makeGetSchemeChannels.